### PR TITLE
Add VSCode devcontainer config

### PR DESCRIPTION
**Pull request by GeorgeLyon**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [ ] Did you state the API impact?
- [ ] Did you specify the code generation impact?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

Development ease-of-use improvement.

#### API Impact

None

#### Backend Code Generation Impact

None

#### Desired Merge Strategy

Squash

#### Release Notes

Added standard config for VSCode dev containers that can get developers up and running with just Docker and VSCode. To use, simply open the repo in VSCode and choose "Rebuild and Reopen in Container".

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x` or `3.6.x` depending on impact, API modification or big change: `5.0.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  make install
+
+# Install GraalVM
+# This downloads all relevant GraalVM architectures at once, mostly because $TARGETARCH values don't map exactly to the release URLs. Since we're optimizing for developer experience here and not image size, this is OK
+# GraalVM release links can be found here: https://github.com/graalvm/graalvm-ce-builds/releases
+ADD https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.3.1/graalvm-ce-java17-linux-aarch64-22.3.1.tar.gz /graalvm/tarballs/arm64.tar.gz
+ADD https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.3.1/graalvm-ce-java17-linux-amd64-22.3.1.tar.gz /graalvm/tarballs/amd64.tar.gz
+RUN tar -xzf /graalvm/tarballs/$TARGETARCH.tar.gz -C /graalvm --strip-components=1
+ENV JAVA_HOME=/graalvm
+ENV PATH=$JAVA_HOME/bin/:$PATH
```
Comment (jackkoenig) [inline .devcontainer/Dockerfile:32]: Why do we need both architectures? I thought the point of Docker was that you can run everything as if it were running on the Linux in the container.

**Comment by sequencer** [comment]
I personally using nix as the development environment, which provides a better support to multi-arch, if using `devcontainer`, there is also a [support](https://github.com/xtruder/nix-devcontainer) for which. WDYT

**Comment by sequencer** [comment]
> If you can get it to work, that is great!

IIRC, I did some black magic on mill bsp. I will check that after back to my office.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (jackkoenig) [reply .devcontainer/Dockerfile:32]: I see, nice!

**Review by jackkoenig — APPROVED** [review]
nix debate aside, I think it's reasonable to merge this.

I think it would also be cool if we could evolve this into a Docker container for CI and for Chisel users as well. In any case, this is useful as is.

**Comment by jackkoenig** [comment]
> Any clues on what the merge failure is about? It says something about me not being a user. I have already contributed to Chisel, so not sure what is happening.

I think there must be a Mergify bug. There is a [Premium Plan Feature](https://docs.mergify.com/actions/queue/) to impersonate Github users when updating pull requests, but I'm pretty sure we're just using the free open-source edition of Mergify and it uses it's own account to update PRs in this repo (eg. see https://github.com/chipsalliance/chisel/pull/3056). If we _were_ able to and using that feature, then the docs say "The user account must have already been
logged in Mergify dashboard once and have write or maintain permission.", so it makes sense it would fail for you, but obviously this shouldn't apply.

Anyway, I'll just update the branch manually--we also should consider not requiring PRs to be up-to-date. I wish we could require PRs to be "up-to-date-ish", like CI has been run in the last 3 days or something.

**Comment by jackkoenig** [comment]
Github has merge queues now, maybe that's a better answer to reduce CI churn while still testing that PRs are up-to-date.

**Pull request merged**